session.go

```diff
@@ -73,6 +69,8 @@ type session struct {
 	// closeChan is used to notify the run loop that it should terminate.
 	// If the value is not nil, the error is sent as a CONNECTION_CLOSE.
 	closeChan chan *qerr.QuicError
+	// the error this session was closed with
+	closeErr  error
```

Comment:
Sure. Right now we have two break statements in the run loop, and at both points we just read the error from the chan. Can we not simply change the breaks to returns of the error? Then we wouldn't need this additional closeErr you're adding here.